include/utility/json_utility.hpp

```diff
@@ -770,21 +712,25 @@ inline std::vector<types::GpioPollingParameters>
         throw std::runtime_error("Missing frus section in system config JSON");
     }
 
-    std::vector<types::GpioPollingParameters> l_pollingRequiredFrusParamsList;
+    std::vector<std::string> l_gpioPollingRequiredFrusList;
 
     for (const auto& l_fru : i_sysCfgJsonObj["frus"].items())
     {
         const auto l_fruPath = l_fru.key();
 
         try
         {
-            types::GpioPollingParameters l_pollingParameters =
-                jsonUtility::getGpioPollingParameters(i_sysCfgJsonObj,
-                                                      l_fruPath);
-
-            if (!std::get<0>(l_pollingParameters).empty())
+            if (isPollingRequired(i_sysCfgJsonObj, l_fruPath))
             {
-                l_pollingRequiredFrusParamsList.push_back(l_pollingParameters);
+                if (i_sysCfgJsonObj["frus"][l_fruPath]
```

Comment:
IMO, This API need not check all the nested flags as it does not get the parameters any more.
This API just need to check if "isPollingRequired" is true for a path, just insert it into the list.

Reply:
pollingRequired is not only for gpio right. later we may have differenent tags under pollingRequired and that may not require gpio polling

Then is it required to instantiate event handler object for that fru

Reply:
Okay, agree to that.